### ISSUE TITLE
Provide a way to validate kubelet identity on AKS

### ISSUE THREAD

**Issue by orltom** [spiffe/helm-charts-hardened#240]
Hello everyone,

We are encountering an issue with SPIRE while utilizing it via the Helm Chart on Azure AKS, specifically when the `skipKubeletVerification` flag is disabled.

The error we're encountering is as follows:
```
time="2024-02-06T17:28:29Z" level=error msg="Failed to collect all selectors for PID" error="workload attestor \"k8s\" failed: rpc error: code = Internal desc = workloadattestor(k8s): unable to perform request: Get \"https://127.0.0.1:10250/pods\": x509: certificate signed by unknown authority" pid=8747 subsystem_name=workload_attestor
```

It appears that Azure generates separate **self-signed** certificates for the kubelet and kube-api server. Here are some details regarding the certificates:

#### Kube API Certificate
```
> kubectl config view --raw --minify --flatten -o jsonpath='{.clusters[0].cluster.certificate-authority-data}' | base64 --decode | openssl x509 -text -noout | head -n12

Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            46:01:92:5f:0a:b4:5e:18:2a:3b:30:28:20:79:d0:24
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: CN=ca
        Validity
            Not Before: Feb  1 13:35:35 2024 GMT
            Not After : Feb  1 13:45:35 2054 GMT
        Subject: CN=ca
```

#### Spire Agent Certificate 
via Service Account and mounted on /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
```
> kubectl get configmaps -n spire kube-root-ca.crt -o json | jq -r '.data[]' |  openssl x509 -text -noout | head -n12

Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            46:01:92:5f:0a:b4:5e:18:2a:3b:30:28:20:79:d0:24
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: CN=ca
        Validity
            Not Before: Feb  1 13:35:35 2024 GMT
            Not After : Feb  1 13:45:35 2054 GMT
        Subject: CN=ca
        Subject Public Key Info:
```

#### Kubelet Certificate
via  Ephemeral Container and OpenSSL CMD
```
>> kubectl debug -it -n spire spire-agent-XYZ --image=ubuntu --target=spire-agent
>> openssl s_client -showcerts -connect 127.0.0.1:10250 2>/dev/null </dev/null | sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' | openssl x509 -text -noout | head -n12

Certificate:
	Data:
    	Version: 3 (0x2)
    	Serial Number:
        		36:3c:9e:af:fe:d3:55:b8:79:16:0a:50:5d:b8:75:1d:e9:12:5c:0b
    	Signature Algorithm: sha256WithRSAEncryption
    	Issuer: CN = aks-system-29976880-vmss000000
    	Validity
        	Not Before: Feb  7 09:07:17 2024 GMT
        	Not After : Feb  2 09:07:17 2044 GMT
    	Subject: CN = aks-system-29976880-vmss000000
    	Subject Public Key Info:

```

Has anyone encountered similar issues on Azure? If my assumption is correct and the spire-agent is unable to validate the kubelet server certificate due to missing certificate, a possible solution could be that Helm Chart provide the kubelet certificate to the spire-agent via a `hostPath` volume mount and set the path via the `kubelet_ca_path` parameter. This approach may allow the spire-agent to validate identity of the kubelet.

**Comment by kfox1111** [comment]
Was looking into this a bit. There are some things to consider for a solution.... 
* The cert may be rotated so mounting just the file in is probably not the right thing to do as it couldn't get updates.
* At least in some of the configurations I've seen, the cert is in the same directory as the key. Which would be bad to have available in the agent container all the time.

For security then, I'm thinking we may want to:
1. hostpath mount the directory the cert is in as a volume
2. Add an emptydir for the cert that is mounted into the agent and the agent reads the cert from
3. Mount the hostpath volume only to a sidecar that mounts the emptydir too, and periodically updates the emptydir with a copy of the cert.

**Comment by orltom** [comment]
Thanks for the detailed consideration of the solution. After re-evaluation, I'm questioning whether the `hostpath` volume approach is advisable, as it could potentially pose security concerns. Nevertheless, I've conducted tests on the same setup in GKE without encountering any issues, likely due to the presence of a certificate chain between the `kubelet` and the `kube-api`. However, the rationale behind generating independent certificates remains unclear to me. Consequently, I'm interested in understanding the approach you would recommend if Spire were deployed outside of the cluster. In Kubernetes clusters under your administrative control, it appears logical to derive the 'kubelet` certificate from the `kube-api` certificate. What are your insights on this?

In summary, would it make sense to raise a ticket on Azure, regardless of the solution of this current issue?
